plugins/airtable/src/api.ts

```diff
@@ -197,7 +195,10 @@ interface FormulaOption {
 interface RollupOption {
     fieldIdInLinkedTable?: string
     recordLinkFieldId?: string
-    result?: AirtableFieldType | null
+    result?: {
+        type: AirtableFieldType
+        options?: AirtableFieldOptions[AirtableFieldType]
+    } | null
```

Comment:
You could define this as the following so that the `options` are inferred correctly based on the `type`

```
type AirtableField<T extends AirtableFieldType> = Extract<AirtableFieldSchema, { type: T }>
type AirtableFieldResult = {
    [K in AirtableFieldType]: Pick<AirtableField<K>, "type" | "options">
}[AirtableFieldType]
```